extensions/cornerstone/src/components/ModalityLoadBadge/ModalityLoadBadge.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ViewportActionButton } from '@ohif/ui-next';
+import { Icons, Tooltip, TooltipTrigger, TooltipContent } from '@ohif/ui-next';
+import { useSystem } from '@ohif/core/src';
+import { useViewportDisplaySets } from '../../hooks/useViewportDisplaySets';
+import { useMeasurementTracking } from '../../hooks/useMeasurementTracking';
+
+/**
+ * ModalityLoadBadge displays the status and actionable buttons for viewports containing
+ * special displaySets (SR, SEG, RTSTRUCT) or when tracking measurements
+ */
+function ModalityLoadBadge({ viewportId }: { viewportId: string }) {
+  const { commandsManager } = useSystem();
+  const { t } = useTranslation('Common');
+  const loadStr = t('LOAD');
+
+  const { isTracked, isLocked } = useMeasurementTracking({ viewportId });
+
+  const { backgroundDisplaySet, overlayDisplaySets } = useViewportDisplaySets(viewportId);
+
+  const [specialDisplaySet, setSpecialDisplaySet] = useState(null);
+
+  const allDisplaySets = useMemo(() => {
+    return [backgroundDisplaySet, ...overlayDisplaySets].filter(Boolean);
+  }, [backgroundDisplaySet, overlayDisplaySets]);
+
+  useEffect(() => {
+    const displaySet = allDisplaySets.find(
+      ds => ds?.Modality === 'SEG' || ds?.Modality === 'RTSTRUCT' || ds?.Modality === 'SR'
+    );
+
+    setSpecialDisplaySet(displaySet || null);
+  }, [allDisplaySets]);
+
+  const statusInfo = useMemo(() => {
+    if (isTracked && !specialDisplaySet) {
+      return {
+        type: 'SR',
+        isHydrated: true,
+        isTracked,
+        displaySet: null,
+        tooltip: 'Currently tracking measurements in this viewport',
+      };
+    }
+
+    if (!specialDisplaySet) {
+      return {
+        type: null,
+      };
+    }
+
+    const type = specialDisplaySet.Modality;
+    const isHydrated = specialDisplaySet.isHydrated || false;
+    let tooltip = null;
+    const isRehydratable = specialDisplaySet.isRehydratable || false;
+
+    if (type === 'SEG') {
```

Comment:
Is there some way to ask the display set for this information so it just gets it as a configuration from the display set?